File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>>mp ;
        vector<vector<string>>ans ; 
        for(auto& ch : strs ){
            string sorted = ch ;
            sort(sorted.begin() , sorted.end()) ;
            mp[sorted].push_back(ch) ; 
        }
        for(auto& i : mp ) {
            ans.push_back(i.second) ;
        }
        return ans ; 
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> mp = new HashMap<>();

        for (String s : strs) {
            char[] cc = s.toCharArray();
            Arrays.sort(cc) ; 
            String sorted = new String(cc) ; 
            if(!mp.containsKey(sorted)) 
            {
                mp.put(sorted , new ArrayList<>()) ; 
            }
            mp.get(sorted).add(s) ;  
        }
        return new ArrayList<>(mp.values()) ; 
    }
}